src/model/guild/member.rs

```diff
@@ -148,28 +129,8 @@ impl Member {
     /// or if a role with a given Id does not exist.
     ///
     /// [Manage Roles]: Permissions::MANAGE_ROLES
-    pub async fn add_roles(
-        &mut self,
-        http: impl AsRef<Http>,
-        role_ids: &[RoleId],
-    ) -> Result<Vec<RoleId>> {
-        self.roles.extend_from_slice(role_ids);
-
-        let mut builder = EditMember::default();
-        builder.roles(&self.roles);
-
-        match http
-            .as_ref()
-            .edit_member(self.guild_id.get(), self.user.id.get(), &builder, None)
-            .await
-        {
-            Ok(member) => Ok(member.roles),
-            Err(why) => {
-                self.roles.retain(|r| !role_ids.contains(r));
-
-                Err(why)
-            },
-        }
+    pub async fn add_roles(&mut self, http: impl AsRef<Http>, role_ids: &[RoleId]) -> Result<()> {
+        self.edit(http, |b| b.roles(role_ids)).await
```

Comment:
This does not add new roles anymore; it instead removes whatever roles the member currently has and replaces them with those in `role_ids`. You will have to append the member's roles with `role_ids`:
```suggestion
        self.edit(http, |b| b.roles(self.roles.iter().chain(role_ids.iter()))).await
```